code/script.js

```diff
@@ -244,56 +248,72 @@ const generateBoard = () => {
   })
 }
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  // Randomly select a person from the characters array and set as the value of the variable called secret.
 }
 
-// This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  // Set charactersInPlay array to be all the characters to start with.
+  generateBoard();
+  setSecret();
 }
 
-// setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  // This function sets the currentQuestion object when you select something in the dropdown.
+  // This variable category stores which option group (category) the question belongs to.
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: questions.value, // value from the input
       category: category,
     }
+
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: questions.value, // value from the input
+      category: category,
+    }
+
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: questions.value,
+      value: true, // boolean value, so always true in the question.
       category: category,
     }
+
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+      currentQuestion = {
+        attribute: questions.value,
+        value: true, // boolean value, so always true in the question.
+        category: category,
+      }
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  // Compares the currentQuestion with the secret person.
+  const secretValue = secret[currentQuestion.attribute];
+
+  if (currentQuestion.value === secretValue) {
+    keep = true
+    filterCharacters(true)
+  } else {
+    keep = false
+    filterCharacters(false)
+  }
```

Comment:
Instead of making the function call for filterCharacter inside the if statement you could make the function call after the if statement and pass the keep variable as a function argument. Example: filterCharacters(keep).